android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt

```diff
@@ -21,6 +25,109 @@ open class LogtoClient(
 
     fun isAuthenticated() = idToken != null
 
+    fun getAccessToken(callback: RetrieveCallback<AccessToken>) =
+        getAccessToken(null, null, callback)
+
+    fun getAccessToken(
+        resource: String?,
+        scope: List<String>?,
+        getAccessTokenCallback: RetrieveCallback<AccessToken>,
+    ) {
+        if (!isAuthenticated()) {
+            getAccessTokenCallback.onResult(LogtoException(LogtoException.Message.NOT_AUTHENTICATED), null)
+            return
+        }
+
+        resource?.let {
+            if (logtoConfig.resource?.contains(it) == false) {
+                getAccessTokenCallback.onResult(
+                    LogtoException(LogtoException.Message.RESOURCE_IS_NOT_GRANTED).apply { detail = it }, null
+                )
+                return
+            }
+        }
+
+        val finalScope = scope ?: logtoConfig.scope
+        if (!logtoConfig.scope.containsAll(finalScope)) {
+            getAccessTokenCallback.onResult(
+                LogtoException(LogtoException.Message.SCOPES_ARE_NOT_ALL_GRANTED).apply {
+                    detail = finalScope.toString()
+                },
+                null
+            )
+            return
+        }
+
+        // Retrieve access token from accessTokenMap
+        val accessTokenKey = buildAccessTokenKey(finalScope, resource)
```

Comment:
wondering wouldn't. cause issue that users have to pass in the exact resource + scope to get the previously granted access_token. 
I.e. resource X with the scope a, b is granted, and user requests for resource X scope b access token. instead of returning the old one, we will send a refresh token request instead.

Reply:
@xiaoyijun we discussed this offline, need to figure out a better way to handle this. for now we can keep as it is. @simeng-li mind to add an issue for discussion?

Reply:
LOG-1451